### PR TITLE
Add log-viewer-webui component with boilerplate Fastify server.

### PR DESCRIPTION
**Pull request by kirkrodrigues**
# Description

To integrate the [log-viewer](https://github.com/y-scope/yscope-log-viewer) with the CLP package, we plan to add a new webapp that consists of a React frontend and a Fastify backend:

* The backend will host the log-viewer.
* The backend will also handle requests to decompress an IR file.
* The frontend will embed the log-viewer in an iframe.
* The frontend will also include controls for jumping between different pages/chunks of the IR file, and jumping between different splits of the log file.

We will integrate this new webapp with the package by embedding it in an iframe in the webui.

This convoluted implementation is for two reasons:

* Integrating the log-viewer with the current webui is non-trivial because the webui is written Meteor which doesn't have good support for web workers.
* The log-viewer isn't yet embeddable as a React component.

This PR starts a new component, `log-viewer-webui`, which contains this webapp. The webapp currently only contains a boilerplate Fastify app.

Note that we haven't yet added the new component to the linting workflow since that requires downloading a new node version which would make the PR a bit bigger.

# Validation performed

* Validated `npm lint:check` passes
* Validated `npm test` passes.

### CODE REVIEW

File: components/log-viewer-webui/server/src/main.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        production: true,
+        test: false,
+    };
+    const server = await app({
+        logger: envToLogger[process.env.NODE_ENV] ?? true,
+    });
+
+    try {
+        const envVars = parseEnvVars();
+        await server.listen({host: envVars.HOST, port: parseInt(envVars.PORT, 10)});
```
Comment (junhaoliao) [inline components/log-viewer-webui/server/src/main.js:55]: Do we want to use `Number()` for string-to-decimal conversions? 

Not sure if we had this discussion before. I'm at fault introducing this `parseInt` convention. There's no difference between the two approaches except:
1. Number() requires the arguments to be strictly a number, while `parseInt(,10)` attempt to parse an int from the argument with best effort. e.g. Number("123abc") -> NaN; parseInt("123abc", 10) -> 123.
2. For an empty string, Number("") returns 0, whereas parseInt("", 10) returns NaN.

```suggestion
        await server.listen({host: envVars.HOST, port: Number(envVars.PORT)});
```

File: components/log-viewer-webui/server/src/main.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        HOST, PORT,
+    } = process.env;
+    const envVars = {
+        HOST, PORT,
+    };
+
+    // Check for mandatory environment variables
+    for (const [key, value] of Object.entries(envVars)) {
+        // eslint-disable-next-line no-undefined
+        if (undefined === value) {
```
Comment (junhaoliao) [inline components/log-viewer-webui/server/src/main.js:28]:
```suggestion
        if ("undefined" === typeof value) {
```

File: components/log-viewer-webui/server/src/main.js (hunk 1 above)
Comment (kirkrodrigues) [reply components/log-viewer-webui/server/src/main.js:55]: Ah sorry, forgot about `Number` amongst all the Fastify stuff.

**Review by junhaoliao — APPROVED** [review]
Looks good to me. 

I verified all routes with Postman and also ensured all npm scripts added in package.json work, including the lint and test ones. The docs/srv/dev-guide md looks good but we do need to publish the site when we merge, or the docs.yscope.com link in components/log-viewer-webui/README.md won't work.

For the env files, we can create a `.env.local` file for entering debug config without git tracking it. That can be done in another PR, preferably after we enable the server in the CLP package / Taskfile.

I agree with the PR title as the squashed commit message.

**Pull request merged**